srcs/Mesh2D.hpp

```diff
@@ -0,0 +1,100 @@
+ * \brief Represents an edge.
+ */
+struct Edge
+{
+    int edgeTag;                        /**< 1D element tag of the edge*/
+
+    std::vector<double> determinant1D;  /**< Determinant of the variable change for Gauss integration,
+                                            evaluated at each Gauss point*/
+
+    std::pair<int, int> nodeTags;       /**< Node tags for each point of the edge*/
+};
+
+/**
+ * \struct Element2D
+ * \brief Represents a 2D element.
+ */
+struct Element2D
+{
+    int elementTag;                     /**< element tag of the element*/
+    int elementType2D;                  /**< 2D type of the element*/
+    int elementType1D;                  /**<12D type of the element*///Store it only once
+
+    std::vector<double> determinant2D;  /**< Determinant of the variable change for Gauss integration,
+                                             evaluated at each Gauss point*/
+    std::vector<double> jacobian2D;     /**< Jacobian of the variable change for Gauss integration,
+                                             evaluated at each Gauss point*/
+
+    std::vector<Edge> edges;            /**< List of edge which compose the element */
+    std::vector<std::pair<double, double>> edgesNormal;     /**< List of edge normal point outwards the element*/
+};
+
+/**
+ * \struct Entity2D
+ * \brief Represents a 2D entity.
+ */
+struct Entity2D
+{
+    int entityTag2D;                    /**< Tag of the 2D entity*/
+    int entityTag1D;                    /**< Tag of the 1D entity linked to this 2D entity*/
+
+    std::vector<Element2D> elements;    /**< List of the elements inside the entity*/
+
+    std::map<int, std::vector<int>> elementTags2D;      /**< Tag of the element inside the entity per element type */
+    std::map<int, std::vector<int>> nodesTags2D;        /**< Tag of the nodes inside the entity per element type */
+    std::map<int, std::vector<int>> nodesTagsPerEdge2D; /**< Tag of the nodes per edge inside the entity per element type */
+};
+
+/**
+ * \struct ElementProperty
+ * \brief Store the different properties of a certain element type.
+ */
+struct ElementProperty
+{
+    std::string name;   /**< Name of the element type */
+    int dim;            /**< Dimension of the element type */
+    int order;          /**< Order of the element type */
+    int numNodes;       /**< Number of nodes of the element type */
+
+    std::vector<double> paramCoord;
+    std::vector<double> basisFunc;      /**< Basis functions evaluated at each Gauss point in the reference axis */
+    std::vector<double> basisFuncGrad;  /**< Basis functions gradient evaluated at each Gauss point in the reference axis */
+    std::vector<double> intPoints;      /**< Integration points for Gauss integration */
+    int numComp;
+};
+
+/**
+ * \struct Mesh2D
+ * \brief Represents a 2D mesh.
+ */
+struct Mesh2D
+{
+    std::map<int, ElementProperty> elementProperties2D; /**< Store element properties for each 2D element type */
```

Comment:
Why do you use the type map?

Reply:
You can access a value in a map like in a vector, but in a vector the index must be from 0 to x, while in a map, it can be whatever you want and you are not forced to add the index 0 before 1. Since we do not know what can be eleType, and which element types there are in the mesh (imagine 2 and 5), a map is better here (with a vector you should have push_back index 0, 1, 3, 4).